nova/virt/vmwareapi/vmops.py

```diff
@@ -2014,10 +2007,47 @@ def finish_migration(self, context, migration, instance, disk_info,
         """Completes a resize, turning on the migrated instance."""
         vm_ref = vm_util.get_vm_ref(self._session, instance)
 
-        # 5. Update ephemerals if necessary
+        flavor = instance.flavor
+        boot_from_volume = compute_utils.is_volume_backed_instance(context,
+                                                                   instance)
+        reattach_volumes = False
+        # 2. Relocate the VM if necessary
+        # If the dest_compute is different from the source_compute, it means we
+        # need to relocate the VM here since we are running on the dest_compute
+        if migration.source_compute != migration.dest_compute:
+            self._detach_volumes(instance, block_device_info)
+            reattach_volumes = True
+            LOG.debug("Relocating VM for migration to %s",
+                      migration.dest_compute, instance=instance)
+            self._relocate_vm(vm_ref, context, instance, network_info,
+                              image_meta)
+            # Refresh vm_ref
+            vm_ref = vm_util.get_vm_ref(self._session, instance)
+            vm_util.update_cluster_placement(self._session, context,
+                                             instance, self._cluster, vm_ref)
+        self._update_instance_progress(context, instance,
+                                       step=2,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+        # 3.Reconfigure the VM and disk
+        self._resize_vm(context, instance, vm_ref, flavor, image_meta)
+        if not boot_from_volume and resize_instance:
+            vmdk = vm_util.get_vmdk_info(self._session, vm_ref,
+                                         uuid=instance.uuid)
+            self._resize_disk(instance, vm_ref, vmdk, flavor)
+        self._update_instance_progress(context, instance,
+                                       step=3,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+
+        # 4. Purge ephemeral and swap disks
+        self._remove_ephemerals_and_swap(vm_ref)
+        self._update_instance_progress(context, instance,
+                                       step=4,
+                                       total_steps=RESIZE_TOTAL_STEPS)
+        # 5. Update ephemerals and attach volumes if necessary
         self._resize_create_ephemerals_and_swap(vm_ref, instance,
                                                 block_device_info)
-
+        if reattach_volumes:
+            self._attach_volumes(instance, block_device_info)
```

Comment:
I'd make this step 6, as it can theoretically take some time if the datastores are not shared between hosts.